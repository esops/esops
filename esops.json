 [
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-node",
   "github:sartaj/esops-ts#2019-10-18-8 --subdir ts-ramda",
   "./package.extend.json -o ./package.json --merge"
 ]