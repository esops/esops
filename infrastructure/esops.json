{
  "stack": ["github:sartaj/typescript-node-stack/stack#0.1.0", "./"],
  "destination": "../"
}
