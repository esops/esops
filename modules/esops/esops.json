<pipe-me />
<target-react-native-web module={'@my/app-navigation'} />
<semantic-design />