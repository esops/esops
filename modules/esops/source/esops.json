[
  "npm:sartaj/typescript-target-stack",
  "http:",
  "https:",
  "git:github.com:sartaj/typescript-stack.git",
  "github:sartaj/typescript-stack"
]
