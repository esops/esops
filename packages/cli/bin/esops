#!/usr/bin/env node

require('@esops/language-babel/node')

// run the CLI with the current process arguments
require('../')(process.argv)
