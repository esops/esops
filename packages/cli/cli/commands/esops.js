// eslint-disable-next-line

const path = require('path')
const target = require('@esops/target-react-native-web').default
// const publishGHPages = require('@esops/publish-github-pages')
const log = require('@esops/logger')

const messages = require('../messages')

const devStaticGithub = async context => {
  const { parameters, cwd } = context

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target({ devMode: true, cwd })
      break
    case 'help':
    default:
      log.md(messages.webAvailableCommands)
  }
}

const shipStaticGithub = async context => {
  const { parameters, cwd } = context

  const distribution = {
    devMode: false,
    buildPath: path.join(cwd, './.esops/target/web/dist')
  }

  switch (parameters.second) {
    case 'web':
    case 'browser':
    case 'github-pages':
      await target(distribution)
      await publishGHPages({ ...context, ...distribution })
      break
    case 'help':
    default:
      log.md('For `esops ship`, available commands are `github-pages`.')
  }
}

// eslint-disable-next-line
module.exports = {
  run: async context => {
    const { parameters } = context

    switch (parameters.first) {
      case 'dev':
        devStaticGithub(context)
        break
      case 'ship':
        shipStaticGithub(context)
        break
      case 'help':
      case undefined:
        await log.mdFile(path.join(__dirname, '../../docs/welcome.md'))
        break
      default:
        log.md('For `esops`, available commands are `dev, ship, & help`.')
    }
  }
}
