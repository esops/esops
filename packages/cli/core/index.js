import conf from 'conf';

const connected = (callback) => {
  callback(features);
};

export default () => {
  const features = {
    storage
  }

  return (callback) => {
    callback(features)
  }
}