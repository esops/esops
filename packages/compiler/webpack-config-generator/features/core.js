/* eslint-disable import/prefer-default-export */

import webpack from 'webpack';
import CleanWebpackPlugin from 'clean-webpack-plugin';

export const paths = ({ buildPath, buildDir, entryPath, libraryName }) => ({
  entry: {
    main: ['babel-polyfill', entryPath].map(require.resolve)
  },

  output: {
    filename: `${libraryName}.[name].[chunkhash].js`,
    library: [libraryName],
    libraryTarget: 'umd',
    path: buildPath,
  },

  devServer: {
    contentBase: buildDir,
  },

  // Paths that can resolve in the require/import statements
  resolve: {
    extensions: ['.js', '.json', '.jsx', '.ts', '.tsx'],
  },

  // Run the webpack that is within this packages context.
  context: __dirname,
});

export const plugins = ({ buildPath }) => ([

  new CleanWebpackPlugin([buildPath], { allowExternal: true }),

  new webpack.optimize.CommonsChunkPlugin({
    name: 'vendor',
    minChunks: Infinity,
  }),

]);
