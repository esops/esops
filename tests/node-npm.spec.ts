import * as path from 'path'
import * as spawn from 'await-spawn'

import {Try} from 'riteway'
import * as prompts from 'prompts'
import {withSnapshots} from './test-utils/withSnapshots'
import {withTempDir} from './test-utils/withTempDir'
import {
  getSortedFilePaths,
  getFileContents,
  getJsonContents,
  cleanErrorString
} from './test-utils/fs-utils'

import esops from '../interfaces/main'
import {MOCK_STACKS} from '../core/examples'

const describe = withSnapshots(__filename)

describe('esops() node+npm features', async assert => {
  await withTempDir(
    __dirname,
    MOCK_STACKS['basic-ignore-files'],
    async destination => {
      await esops({destination})

      await assert({
        given: 'an included .npmignore',
        should: 'have updated .npmignore with generated file paths',
        snap: getFileContents(path.join(destination, '.npmignore'))
      })
    }
  )

  await withTempDir(__dirname, MOCK_STACKS['basic-package-json'], async cwd => {
    await esops({cwd})

    await assert({
      given: 'a minimal package with package.json',
      should: 'generate basic template in cwd',
      snap: getSortedFilePaths(cwd)
    })

    await assert({
      given: 'an included .gitignore',
      should: 'have updated .gitignore with generated paths',
      snap: getFileContents(path.join(cwd, '.gitignore'))
    })

    await assert({
      given: 'a minimal package.json',
      should: 'have contents in package.json',
      snap: getJsonContents(path.join(cwd, 'package.json'))
    })
  })

  await withTempDir(__dirname, MOCK_STACKS['basic-node-module'], async cwd => {
    await spawn(`npm`, ['install'], {cwd})
    await esops({cwd})
    await assert({
      given: 'minimal stack from node module',
      should: 'generate basic template in cwd from node_module',
      snap: getSortedFilePaths(cwd)
    })
  })
})
