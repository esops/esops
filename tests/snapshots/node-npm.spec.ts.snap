module.exports[`Describe esops() node+npm features | Given an included .npmignore: should have updated .npmignore with generated file paths`] = `

/src/**
### ESOPS AUTO GENERATED BEGIN ###
/.vscode/settings.json
/src/stores/stores-architecture.md
/tsconfig.json
### ESOPS AUTO GENERATED END ###
`

